### DISCUSSION TITLE
How do i use inner objects in @entity class

### DISCUSSION THREAD

**Discussion by thq919** [vitusortner/floor#675] [Q&A] [answered]
```
class CartEntity {
  String delivery;
  @primaryKey
  String id;
  int total;
List<CartItemEntity> basket;

}
```
More shortly such problem, swears that is not able to store in CartItemEntity array.
How can I make it possible?

error
```
[SEVERE] floor_generator:floor_generator on lib/core/db/_database.dart:

Column type is not supported for List<CartItemEntity>.
package:shopest/features/cart/domain/entities/cart_entity.dart:11:24
   ╷
11 │   List<CartItemEntity> basket;
   │                        ^^^^^^
   ╵
```

**Comment by dkaera** [comment, marked as answer]
For the inner objects, you need to use [TypeConverters](https://pinchbv.github.io/floor/type-converters/) because the entity is like a table view where each declared field represents a column name with a primitive type and the field's value is the value in the cell.
More information about supported types is [here](https://pinchbv.github.io/floor/entities/#supported-types)

**Reply by thq919** [reply]
Do you have any examples of using that feature?
Or, i heard, foreign key is kinda exit from that situation.
Maybe examples of using foreign keys.
Didn't found any from floor\examples, or mentioning that in documentation.
Thanks for previous answer, really helped, but now im little bit stuck at encode decode process.

**Reply by dkaera** [reply]
Good point, current project does not include the integration example. Will update it soon.

**Reply by dkaera** [reply]
@thq919 I've created [PR](https://github.com/pinchbv/floor/pull/678), please see an example of using TypeConverters
